### ISSUE TITLE
Gallery App is built and tested for iOS and Android

### ISSUE THREAD

**Issue by sethladd** [flutter/flutter#1600]
- [x] On dashboard
- [x] Android phone
- [x] iOS phone (https://github.com/flutter/flutter/issues/4359)

Testing and verification can include:
- Testing app on Nexus 5
- Testing app on iPhone 6
- Performance
  - Startup
  - Animations (TODO, don't have the infra for this quite yet)
  - Scrolling (grid, table, lists, calendar)
- Correctness
  - smoke test for automated bots
  - Manual testing plan

**Comment by eseidelGoogle** [comment]
At time of writing, none of Flutter's framework supports anything other than phones in portrait organization.  Various widgets/layouts will need updating for tablet.  @Hixie @krisgiesing

**Comment by eseidelGoogle** [comment]
We build for both iOS and Android on the bots, we briefly ran `flutter drive` in the iOS simulator, but we definitely have more we could do here.

FYI @chinmaygarde @yjbanov

**Comment by yjbanov** [comment]
We now run the benchmarks for the gallery app regularly. The startup benchmark is already on the dashboard. @HansMuller wrote a benchmark for screen transitions. I'm adding it to the dashboard now.

Can we reduce the scope of the issue to Android phones only for now? Otherwise, to fix it one would find themselves shaving a herd of yaks in no time. Instead, I propose that as we expand our test device matrix to iOS and then tablets we just take all existing tests and benchmarks with us.

With a reduced scope I'd be willing to take this issue.

**Comment by eseidelGoogle** [comment]
Sold.  Lets dump the tablet bits for now.

**Comment by yjbanov** [comment]
We now have the screen transition tests running and on the dashboard too. The remaining parts are to make this all run on iOS. This is currently blocked by a number of other issues, so I'll put this on hold until we're unblocked.

Blockers:

- https://github.com/flutter/flutter/issues/3947
- https://github.com/flutter/flutter/issues/3754

**Comment by sethladd** [comment]
Thanks for the links to the blockers! I've updated their milestones to indicate they are part of the Gallery milestone.

**Comment by chinmaygarde** [comment]
I believe this is resolved once https://github.com/flutter/flutter/pull/4575 lands.

**Comment by yjbanov** [comment]
@chinmaygarde not quite. I still need to get the tests running on the dashboard box. But #4575 gets us very close.

**Comment by sethladd** [comment]
Any updates on testing on real iOS hardware?

**Comment by yjbanov** [comment]
We now build and test on both Android and iOS. I even had iOS briefly passing until we regressed again. Not sure if we should mark it as fixed and open a separate issue to fix the iOS build or wait until our iOS tests are green.

**Comment by sethladd** [comment]
Don't think we should close this yet. We aren't able to verify gallery builds via our tests on iOS hardware yet, which is the intent of this issue.

**Comment by yjbanov** [comment]
Closing this. Android is all done, and we _are_ running some of iOS. I created multiple bite-sized issues in the [Critical Metrics project](https://github.com/flutter/flutter/projects/6) to have iOS catch up:

https://github.com/flutter/flutter/issues/7074 - gallery perf/smoke test
https://github.com/flutter/flutter/issues/7075 - gallery start-up test
https://github.com/flutter/flutter/issues/7076 - scroll test
